pkg/proxy/iptables/proxier.go

```diff
@@ -572,45 +590,56 @@ func (proxier *Proxier) OnEndpointsSynced() {
 	proxier.syncProxyRules(syncReasonEndpoints)
 }
 
-// Convert a slice of api.Endpoints objects into a map of service-port -> endpoints.
-func buildNewEndpointsMap(allEndpoints endpointsMap, curMap proxyEndpointMap, hostname string) (newMap proxyEndpointMap, hcEndpoints map[types.NamespacedName]int, staleSet map[endpointServicePair]bool) {
+func (proxier *Proxier) mergeEndpointsMap(endpointsMap proxyEndpointsMap) {
+	for svcPort := range endpointsMap {
+		proxier.endpointsMap[svcPort] = endpointsMap[svcPort]
+	}
 
-	// return values
-	newMap = make(proxyEndpointMap)
-	hcEndpoints = make(map[types.NamespacedName]int)
-	staleSet = make(map[endpointServicePair]bool)
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ExternalTrafficLocalOnly) {
```

Comment:
my inclination is to move this into the sync loop rather than caching it in the proxier object.  It's a bit more compute, but once the sync is properly throttled, shouldn't be a problem?  Thoughts?

That would remove this block from merge/unmerge, making these trivial functions.